crates/viewer/re_space_view_dataframe/src/space_view_class.rs

```diff
@@ -166,20 +221,322 @@ mode sets the default time range to _everything_. You can override this in the s
+                })
+                .collect::<Vec<_>>();
+
+            let selected_component_columns = schema
+                .iter()
+                .filter_map(|column| match column {
+                    ColumnDescriptor::Component(desc) => {
+                        Some(components::ComponentColumnSelector::new(
+                            &desc.entity_path,
+                            desc.component_name,
+                        ))
+                    }
+                    _ => None,
+                })
+                .collect::<Vec<_>>();
+
+            (selected_time_columns, selected_component_columns)
+        } else {
+            (
+                property.component_array_or_empty::<components::TimelineName>()?,
+                property.component_array_or_empty::<components::ComponentColumnSelector>()?,
+            )
+        };
+
+    for action in actions {
+        match action {
+            HideColumnAction::HideTimeColumn { timeline_name } => {
+                selected_time_columns
+                    .retain(|name| name != &components::TimelineName::from(timeline_name.as_str()));
+            }
+
+            HideColumnAction::HideComponentColumn {
+                entity_path,
+                component_name,
+            } => {
+                let blueprint_component_descriptor =
+                    components::ComponentColumnSelector::new(&entity_path, component_name);
+                selected_component_columns.retain(|desc| desc != &blueprint_component_descriptor);
+            }
+        }
+    }
+
+    property.save_blueprint_component(ctx, &selected_time_columns);
+    property.save_blueprint_component(ctx, &selected_component_columns);
+
+    Ok(())
+}
+
+fn column_visibility_ui(
+    ctx: &ViewerContext<'_>,
+    ui: &mut egui::Ui,
+    space_view_id: SpaceViewId,
+    schema: &[ColumnDescriptor],
+    query_timeline_name: &TimelineName,
+) -> Result<(), SpaceViewSystemExecutionError> {
+    let property = ViewProperty::from_archetype::<archetypes::DataframeVisibleColumns>(
+        ctx.blueprint_db(),
+        ctx.blueprint_query,
+        space_view_id,
+    );
+
+    let menu_ui = |ui: &mut egui::Ui| -> Result<(), SpaceViewSystemExecutionError> {
+        ui.style_mut().wrap_mode = Some(egui::TextWrapMode::Extend);
+
+        //
+        // All or selected?
+        //
+
+        let mut column_selection_mode = property
+            .component_or_empty::<components::ColumnSelectionMode>()?
+            .unwrap_or_default();
+
+        let changed = {
+            ui.re_radio_value(
+                &mut column_selection_mode,
+                components::ColumnSelectionMode::All,
+                "All",
+            )
+            .changed()
+        } || {
+            ui.re_radio_value(
+                &mut column_selection_mode,
+                components::ColumnSelectionMode::Selected,
+                "Selected",
+            )
+            .changed()
+        };
```

Comment:
The short-circuting of `||` lead to the second `radio_value` becoming invisible if the first one was clicked

```suggestion
        let changed = (
            ui.re_radio_value(
                &mut column_selection_mode,
                components::ColumnSelectionMode::All,
                "All",
            ) |
            ui.re_radio_value(
                &mut column_selection_mode,
                components::ColumnSelectionMode::Selected,
                "Selected",
            )
        ).changed();
```